### ISSUE TITLE
Error when executing jmetermetrics

### ISSUE THREAD

**Issue by uasetum** [adiralashiva8/jmeter-metrics#1]
There is a jmeter log file - result.jtl in current folder. Every time I execute the script (with or w/o options) I have the error below.

Python 3.8.0 + virtual environment

(p380) d:\Repos\jmeter-metrics\results>jmetermetrics
Traceback (most recent call last):
  File "d:\Repos\jmeter-metrics\p380\Scripts\jmetermetrics-script.py", line 11, in <module>
    load_entry_point('jmeter-metrics==1.0.4', 'console_scripts', 'jmetermetrics')()
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\runner.py", line 62, in main
    generate_report(args)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\jmetermetrics.py", line 53, in generate_report
    df = pd.concat(df_from_each_file, ignore_index=True)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\core\reshape\concat.py", line 271, in concat
    op = _Concatenator(
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\core\reshape\concat.py", line 326, in __init__
    objs = list(objs)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\jmetermetrics.py", line 52, in <genexpr>
    df_from_each_file = (pd.read_csv(f) for f in output_names)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 676, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 454, in _read
    data = parser.read(nrows)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 1133, in read
    ret = self._engine.read(nrows)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 2037, in read
    data = self._reader.read(nrows)
  File "pandas\_libs\parsers.pyx", line 860, in pandas._libs.parsers.TextReader.read
  File "pandas\_libs\parsers.pyx", line 875, in pandas._libs.parsers.TextReader._read_low_memory
  File "pandas\_libs\parsers.pyx", line 929, in pandas._libs.parsers.TextReader._read_rows
  File "pandas\_libs\parsers.pyx", line 916, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas\_libs\parsers.pyx", line 2071, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 1 fields in line 9, saw 3

**Comment by adiralashiva8** [comment]
Seems issue with python compatibility or invalid arguments

I have tested this library in python3.7

Please try one more time with option: `jmetermetrics --output test.jtl`

Could you please test in python 3.7 & argument as mentioned above

In mean time i will work on this issue

**Comment by uasetum** [comment]
Same result

(p380) d:\Repos\jmeter-metrics\results>jmetermetrics --output result.jtl
Traceback (most recent call last):
  File "d:\Repos\jmeter-metrics\p380\Scripts\jmetermetrics-script.py", line 11, in <module>
    load_entry_point('jmeter-metrics==1.0.4', 'console_scripts', 'jmetermetrics')()
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\runner.py", line 62, in main
    generate_report(args)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\jmetermetrics.py", line 53, in generate_report
    df = pd.concat(df_from_each_file, ignore_index=True)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\core\reshape\concat.py", line 271, in concat
    op = _Concatenator(
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\core\reshape\concat.py", line 326, in __init__
    objs = list(objs)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\jmeter_metrics\jmetermetrics.py", line 52, in <genexpr>
    df_from_each_file = (pd.read_csv(f) for f in output_names)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 676, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 454, in _read
    data = parser.read(nrows)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 1133, in read
    ret = self._engine.read(nrows)
  File "d:\repos\jmeter-metrics\p380\lib\site-packages\pandas\io\parsers.py", line 2037, in read
    data = self._reader.read(nrows)
  File "pandas\_libs\parsers.pyx", line 860, in pandas._libs.parsers.TextReader.read
  File "pandas\_libs\parsers.pyx", line 875, in pandas._libs.parsers.TextReader._read_low_memory
  File "pandas\_libs\parsers.pyx", line 929, in pandas._libs.parsers.TextReader._read_rows
  File "pandas\_libs\parsers.pyx", line 916, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas\_libs\parsers.pyx", line 2071, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 1 fields in line 10, saw 3

**Comment by adiralashiva8** [comment]
@uasetum Shall i reopen bug if its not working? I want to troubleshoot this issue

**Comment by uasetum** [comment]
Would be nice solve this issue. Do you need jtl data file?

**Comment by uasetum** [comment]
It seems I found the issue. I use pipe symbol ('|') as data delimiter

**Comment by uasetum** [comment]
Here is a data sample I have

[result.txt](https://github.com/adiralashiva8/jmeter-metrics/files/4753557/result.txt)

**Comment by adiralashiva8** [comment]
> It seems I found the issue. I use pipe symbol ('|') as data delimiter

Thanks a lot for notifying issue, may be need to provide delimiter option to users to resolve this kind of issue. Will try to support ASAP

**Comment by adiralashiva8** [comment]
@uasetum This got fixed now & changes are in `v1.0.5`. Provided new command line option to specify delimiter `-S` or `--seperator`

please go through following steps to validate

Install latest version: 
```
pip install jmeter-metrics==1.0.5
```

Execute following command to parse csv/jtl file which contains custom delimiter
```
jmetermetrics -O results.txt -S "|"
```

**Comment by uasetum** [comment]
Fixed. Thanks
